Simon-Initiative/oli-torus
# Curriculum refactor + Breadcrumbs

This PR adds curriculum changes and breadcrumb trails based on [this slide deck](https://docs.google.com/presentation/d/1bbADcItf9IPWi1OgKv3u-sNKADZwZOI5RpKskuHg5iY/edit) and [this implementation strategy](https://docs.google.com/document/d/12c57lT9NbrMv3uMIaD2hT1F-FmwlQVWtgRyWrAamTCk/edit#). The container dropdown menu is not supported - there's some discussion needed around that first.

**Note**: requires a `mix deps.get`

**Curriculum**: see [this pull request](https://github.com/Simon-Initiative/oli-torus/pull/600). Comments are addressed.

**Breadcrumb trail**

_How to review_:
- Review code for:
  - `lib/oli_web/live/breadcrumb/`
  - `lib/oli_web/live/curriculum/entries/`
  - `lib/oli_web/live/curriculum/settings/`
  - `lib/oli/resources/numbering.ex`
  - `lib/oli/utils/breadcrumb.ex`
  - Other relevant code is in `lib/oli_web/live/curriculum/container/`, but much of it is the same from the `Curriculum Refactor` PR
- Go to curriculum
  - Add new graded page
    - Click into page, make sure it works
    - Go back to curriculum using the breadcrumb trail at the top
    - Edit settings for page using the curriculum view
    - Reload page with settings menu open, make sure it persists
  - Add new ungraded page
  - Add new container
    - Go into container, make sure the container name displays in the breadcrumb
    - Add a second-level container and then a third-level container
      - Go into third-level container, make sure the breadcrumb trail names at the top are "shortened"
  - Change container name using settings menu
  - Make sure deletion works from settings menu for both pages and containers
